common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -371,4 +390,62 @@ private static void closeQuietly(Closeable c) {
     private NativeLibraryLoader() {
         // Utility
     }
+
+    private static final class NoexecVolumeDetector {
+
+        private static boolean canExecuteExecutable(File file)
+                throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException,
+                       IllegalAccessException {
+            if (PlatformDependent.javaVersion() < 7) {
+                // Assume true because we
+                return true;
+            }
+
+            // If we can already execute, there is nothing to do.
+            if (file.canExecute()) {
+                return true;
+            }
+
+            // On volumes, with noexec set, even files with the executable POSIX permissions will fail to execute.
+            // So, check for executable permissions, then find out if the file can be executed.
+
+            // Everything is done reflectively to prevent the animal-sniffer plugin from complainin.
+            Class<?> filesClass = Class.forName("java.nio.file.Files");
```

Comment:
instead of using reflection to load these resources can we reference the class directly? We just need to be careful about class loading on older JDKs.

```java
private static boolean canExecuteFile(File f) {
  return PlatformDependent.javaVersion() < 7 ? true : Java7FilesUtil. canExecuteFile(f);
}

// In another class file ... Java7FilesUtil.java
final class Java7FilesUtil {
  static boolean canExecuteFile(File f) {
    ...
  }
}
```